src/morphosamplers/spline.py

```diff
@@ -0,0 +1,173 @@
+"""Tooling to fit and sample splines."""
+
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import einops
+import numpy as np
+from psygnal import EventedModel
+from pydantic import PrivateAttr, root_validator, validator
+from scipy.interpolate import splev, splprep
+
+
+class NDimensionalSpline(EventedModel):
+    """Model for multidimensional splines."""
+
+    points: np.ndarray
+    spline_order: int = 3
```

Comment:
```suggestion
    order: int = 3
```

`spline.spline_order` might feel weird, maybe just call it `order`?

Reply:
Good idea. updated